spec/data_structures/linked_list/linked_list_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'spec_helper'
+
+RSpec.describe LinkedList do
+  let(:head_node){ LinkedList::Node.new('A head value') }
+
+	describe '#new' do
+   let(:wrong_class) { Hash.new }
+
+    context 'with node argument' do
```

Comment:
Should be better the `LinkedList::Node` initialization be LinkedList responsability.
We may pass values or any classes as the argument to initialize the head node. Additionally, let us be able to pass any class to `#append` method and leave the Node instantiation up to LinkedList class.